subsys/testsuite/ztest/src/ztest_new.c

```diff
@@ -675,10 +675,6 @@ static int z_ztest_run_test_suite_ptr(struct ztest_suite_node *suite)
 	int fail = 0;
 	int tc_result = TC_PASS;
 
-	if (test_status < 0) {
```

Comment:
I think you found the cause of https://github.com/zephyrproject-rtos/zephyr/issues/60074. @asemjonovs I'm going to approve this PR, but can you verify that replacing this with `test_status != ZTEST_STATUS_OK` fixes the issue?

Reply:
I was not aware of this issue... but I tried myself

without the change
```
-- west build: running target run
[0/1] To exit from QEMU enter: 'CTRL+a, x'[QEMU] CPU: cortex-m3
qemu-system-arm: warning: nic stellaris_enet.0 has no peer
Timer with period zero, disabling
*** Booting Zephyr OS build zephyr-v3.4.0-1607-ga98d2c3aafce ***
Running TESTSUITE semaphore_1cpu
===================================================================
START - test_sem_queue_mutual_exclusion
 PASS - test_sem_queue_mutual_exclusion in 0.193 seconds
===================================================================
START - test_sem_multiple_take_and_timeouts
 PASS - test_sem_multiple_take_and_timeouts in 1.256 seconds
===================================================================
START - test_sem_queue_mutual_exclusion
 PASS - test_sem_queue_mutual_exclusion in 0.193 seconds
===================================================================
START - test_sem_multiple_take_and_timeouts
 PASS - test_sem_multiple_take_and_timeouts in 1.256 seconds
===================================================================
START - test_sem_multiple_take_and_timeouts
 PASS - test_sem_multiple_take_and_timeouts in 1.260 seconds
===================================================================
START - test_sem_queue_mutual_exclusion
 PASS - test_sem_queue_mutual_exclusion in 0.194 seconds
===================================================================
TESTSUITE semaphore_1cpu succeeded
Running TESTSUITE semaphore_1cpu
===================================================================
START - test_sem_queue_mutual_exclusion
 PASS - test_sem_queue_mutual_exclusion in 0.189 seconds
===================================================================
START - test_sem_multiple_take_and_timeouts
 PASS - test_sem_multiple_take_and_timeouts in 1.256 seconds
===================================================================
START - test_sem_multiple_take_and_timeouts
 PASS - test_sem_multiple_take_and_timeouts in 1.260 seconds
===================================================================
START - test_sem_queue_mutual_exclusion
 PASS - test_sem_queue_mutual_exclusion in 0.194 seconds
===================================================================
START - test_sem_multiple_take_and_timeouts
 PASS - test_sem_multiple_take_and_timeouts in 1.256 seconds
===================================================================
START - test_sem_queue_mutual_exclusion
 PASS - test_sem_queue_mutual_exclusion in 0.194 seconds
===================================================================
TESTSUITE semaphore_1cpu succeeded
Running TESTSUITE semaphore_1cpu
===================================================================
START - test_sem_multiple_take_and_timeouts
 PASS - test_sem_multiple_take_and_timeouts in 1.255 seconds
===================================================================
START - test_sem_queue_mutual_exclusion
 PASS - test_sem_queue_mutual_exclusion in 0.194 seconds
===================================================================
START - test_sem_queue_mutual_exclusion
 PASS - test_sem_queue_mutual_exclusion in 0.190 seconds
===================================================================
START - test_sem_multiple_take_and_timeouts
 PASS - test_sem_multiple_take_and_timeouts in 1.256 seconds
===================================================================
START - test_sem_multiple_take_and_timeouts
 PASS - test_sem_multiple_take_and_timeouts in 1.260 seconds
===================================================================
START - test_sem_queue_mutual_exclusion
 PASS - test_sem_queue_mutual_exclusion in 0.194 seconds
===================================================================
TESTSUITE semaphore_1cpu succeeded
Running TESTSUITE semaphore_null_case
===================================================================
TESTSUITE semaphore_null_case succeeded
Running TESTSUITE semaphore_null_case
===================================================================
TESTSUITE semaphore_null_case succeeded
Running TESTSUITE semaphore_null_case
===================================================================
TESTSUITE semaphore_null_case succeeded
Running TESTSUITE semaphore
===================================================================
START - test_k_sem_define
 PASS - test_k_sem_define in 0.001 seconds
===================================================================
START - test_sem_thread2thread
 PASS - test_sem_thread2thread in 0.001 seconds
===================================================================
START - test_sem_multi_take_timeout_diff_sem
 PASS - test_sem_multi_take_timeout_diff_sem in 5.009 seconds
===================================================================
START - test_sem_take_timeout_fails
 PASS - test_sem_take_timeout_fails in 0.550 seconds
===================================================================
START - test_sem_take_no_wait
 PASS - test_sem_take_no_wait in 0.001 seconds
===================================================================
START - test_sem_take_no_wait_fails
 PASS - test_sem_take_no_wait_fails in 0.001 seconds
===================================================================
START - test_k_sem_correct_count_limit
 PASS - test_k_sem_correct_count_limit in 0.001 seconds
===================================================================
START - test_sem_reset
 PASS - test_sem_reset in 0.108 seconds
===================================================================
START - test_sem_give_take_from_isr
 PASS - test_sem_give_take_from_isr in 0.001 seconds
===================================================================
START - test_sem_give_from_thread
 PASS - test_sem_give_from_thread in 0.001 seconds
===================================================================
START - test_sem_multiple_threads_wait
 PASS - test_sem_multiple_threads_wait in 2.038 seconds
===================================================================
START - test_k_sem_init
 PASS - test_k_sem_init in 0.001 seconds
===================================================================
START - test_sem_give_from_isr
 PASS - test_sem_give_from_isr in 0.001 seconds
===================================================================
START - test_sem_measure_timeout_from_thread

    Assertion failed at WEST_TOPDIR/zephyr/tests/kernel/semaphore/semaphore/src/main.c:947: semaphore_test_sem_measure_timeout_from_thread: (1 is non-zero)
Forced error
 FAIL - test_sem_measure_timeout_from_thread in 0.001 seconds
===================================================================
START - test_sem_count_get
 PASS - test_sem_count_get in 0.001 seconds
===================================================================
START - test_sem_take_timeout_isr
 PASS - test_sem_take_timeout_isr in 0.001 seconds
===================================================================
START - test_sem_thread2isr
 PASS - test_sem_thread2isr in 0.001 seconds
===================================================================
START - test_sem_reset_waiting
 PASS - test_sem_reset_waiting in 0.016 seconds
===================================================================
START - test_sem_measure_timeouts
 PASS - test_sem_measure_timeouts in 1.010 seconds
===================================================================
START - test_sem_take_multiple
 PASS - test_sem_take_multiple in 1.010 seconds
===================================================================
START - test_sem_take_timeout_forever
 PASS - test_sem_take_timeout_forever in 0.110 seconds
===================================================================
START - test_sem_take_timeout
 PASS - test_sem_take_timeout in 0.001 seconds
===================================================================
START - test_sem_measure_timeouts
 PASS - test_sem_measure_timeouts in 1.009 seconds
===================================================================
START - test_sem_take_no_wait_fails
 PASS - test_sem_take_no_wait_fails in 0.001 seconds
===================================================================
START - test_sem_take_timeout_forever
 PASS - test_sem_take_timeout_forever in 0.109 seconds
===================================================================
START - test_sem_reset_waiting
 PASS - test_sem_reset_waiting in 0.020 seconds
===================================================================
START - test_k_sem_init
 PASS - test_k_sem_init in 0.001 seconds
===================================================================
START - test_k_sem_correct_count_limit
 PASS - test_k_sem_correct_count_limit in 0.001 seconds
===================================================================
START - test_sem_give_from_thread
 PASS - test_sem_give_from_thread in 0.001 seconds
===================================================================
START - test_sem_give_from_isr
 PASS - test_sem_give_from_isr in 0.001 seconds
===================================================================
START - test_sem_thread2thread
 PASS - test_sem_thread2thread in 0.001 seconds
===================================================================
START - test_sem_take_no_wait
 PASS - test_sem_take_no_wait in 0.001 seconds
===================================================================
START - test_sem_take_multiple
 PASS - test_sem_take_multiple in 1.006 seconds
===================================================================
START - test_sem_thread2isr
 PASS - test_sem_thread2isr in 0.001 seconds
===================================================================
START - test_sem_take_timeout
 PASS - test_sem_take_timeout in 0.001 seconds
===================================================================
START - test_k_sem_define
 PASS - test_k_sem_define in 0.001 seconds
===================================================================
START - test_sem_count_get
 PASS - test_sem_count_get in 0.001 seconds
===================================================================
START - test_sem_measure_timeout_from_thread
 PASS - test_sem_measure_timeout_from_thread in 0.001 seconds
===================================================================
START - test_sem_multiple_threads_wait
 PASS - test_sem_multiple_threads_wait in 2.037 seconds
===================================================================
START - test_sem_take_timeout_fails
 PASS - test_sem_take_timeout_fails in 0.550 seconds
===================================================================
START - test_sem_give_take_from_isr
 PASS - test_sem_give_take_from_isr in 0.001 seconds
===================================================================
START - test_sem_reset
 PASS - test_sem_reset in 0.109 seconds
===================================================================
START - test_sem_multi_take_timeout_diff_sem
 PASS - test_sem_multi_take_timeout_diff_sem in 5.010 seconds
===================================================================
START - test_sem_take_timeout_isr
 PASS - test_sem_take_timeout_isr in 0.001 seconds
===================================================================
START - test_sem_take_no_wait
 PASS - test_sem_take_no_wait in 0.001 seconds
===================================================================
START - test_k_sem_correct_count_limit
 PASS - test_k_sem_correct_count_limit in 0.001 seconds
===================================================================
START - test_sem_take_timeout_isr
 PASS - test_sem_take_timeout_isr in 0.001 seconds
===================================================================
START - test_sem_give_from_thread
 PASS - test_sem_give_from_thread in 0.001 seconds
===================================================================
START - test_sem_measure_timeout_from_thread
 PASS - test_sem_measure_timeout_from_thread in 0.001 seconds
===================================================================
START - test_sem_multi_take_timeout_diff_sem
 PASS - test_sem_multi_take_timeout_diff_sem in 5.005 seconds
===================================================================
START - test_k_sem_define
 PASS - test_k_sem_define in 0.001 seconds
===================================================================
START - test_sem_multiple_threads_wait
 PASS - test_sem_multiple_threads_wait in 2.039 seconds
===================================================================
START - test_sem_give_take_from_isr
 PASS - test_sem_give_take_from_isr in 0.001 seconds
===================================================================
START - test_k_sem_init
 PASS - test_k_sem_init in 0.001 seconds
===================================================================
START - test_sem_take_timeout_forever
 PASS - test_sem_take_timeout_forever in 0.108 seconds
===================================================================
START - test_sem_take_no_wait_fails
 PASS - test_sem_take_no_wait_fails in 0.001 seconds
===================================================================
START - test_sem_take_timeout_fails
 PASS - test_sem_take_timeout_fails in 0.549 seconds
===================================================================
START - test_sem_count_get
 PASS - test_sem_count_get in 0.001 seconds
===================================================================
START - test_sem_take_timeout
 PASS - test_sem_take_timeout in 0.001 seconds
===================================================================
START - test_sem_thread2thread
 PASS - test_sem_thread2thread in 0.001 seconds
===================================================================
START - test_sem_reset_waiting
 PASS - test_sem_reset_waiting in 0.018 seconds
===================================================================
START - test_sem_reset
 PASS - test_sem_reset in 0.110 seconds
===================================================================
START - test_sem_thread2isr
 PASS - test_sem_thread2isr in 0.001 seconds
===================================================================
START - test_sem_give_from_isr
 PASS - test_sem_give_from_isr in 0.001 seconds
===================================================================
START - test_sem_measure_timeouts
 PASS - test_sem_measure_timeouts in 1.009 seconds
===================================================================
START - test_sem_take_multiple
 PASS - test_sem_take_multiple in 1.010 seconds
===================================================================
TESTSUITE semaphore succeeded
Running TESTSUITE semaphore
===================================================================
START - test_sem_give_from_thread
 PASS - test_sem_give_from_thread in 0.001 seconds
===================================================================
START - test_sem_take_timeout_isr
 PASS - test_sem_take_timeout_isr in 0.001 seconds
===================================================================
START - test_sem_thread2thread
 PASS - test_sem_thread2thread in 0.001 seconds
===================================================================
START - test_sem_multiple_threads_wait
 PASS - test_sem_multiple_threads_wait in 2.037 seconds
===================================================================
START - test_k_sem_correct_count_limit
 PASS - test_k_sem_correct_count_limit in 0.001 seconds
===================================================================
START - test_sem_take_no_wait
 PASS - test_sem_take_no_wait in 0.001 seconds
===================================================================
START - test_sem_reset_waiting
 PASS - test_sem_reset_waiting in 0.019 seconds
===================================================================
START - test_sem_reset
 PASS - test_sem_reset in 0.110 seconds
===================================================================
START - test_sem_count_get
 PASS - test_sem_count_get in 0.001 seconds
===================================================================
START - test_k_sem_define
 PASS - test_k_sem_define in 0.001 seconds
===================================================================
START - test_sem_measure_timeout_from_thread
 PASS - test_sem_measure_timeout_from_thread in 0.001 seconds
===================================================================
START - test_sem_thread2isr
 PASS - test_sem_thread2isr in 0.001 seconds
===================================================================
START - test_k_sem_init
 PASS - test_k_sem_init in 0.001 seconds
===================================================================
START - test_sem_take_timeout_forever
 PASS - test_sem_take_timeout_forever in 0.107 seconds
===================================================================
START - test_sem_take_no_wait_fails
 PASS - test_sem_take_no_wait_fails in 0.001 seconds
===================================================================
START - test_sem_take_timeout
 PASS - test_sem_take_timeout in 0.001 seconds
===================================================================
START - test_sem_give_take_from_isr
 PASS - test_sem_give_take_from_isr in 0.001 seconds
===================================================================
START - test_sem_give_from_isr
 PASS - test_sem_give_from_isr in 0.001 seconds
===================================================================
START - test_sem_take_multiple
 PASS - test_sem_take_multiple in 1.007 seconds
===================================================================
START - test_sem_measure_timeouts
 PASS - test_sem_measure_timeouts in 1.010 seconds
===================================================================
START - test_sem_take_timeout_fails
 PASS - test_sem_take_timeout_fails in 0.550 seconds
===================================================================
START - test_sem_multi_take_timeout_diff_sem
 PASS - test_sem_multi_take_timeout_diff_sem in 5.010 seconds
===================================================================
START - test_sem_take_no_wait_fails
 PASS - test_sem_take_no_wait_fails in 0.001 seconds
===================================================================
START - test_sem_take_timeout_isr
 PASS - test_sem_take_timeout_isr in 0.001 seconds
===================================================================
START - test_sem_multi_take_timeout_diff_sem
 PASS - test_sem_multi_take_timeout_diff_sem in 5.008 seconds
===================================================================
START - test_sem_thread2isr
 PASS - test_sem_thread2isr in 0.001 seconds
===================================================================
START - test_sem_reset
 PASS - test_sem_reset in 0.109 seconds
===================================================================
START - test_sem_thread2thread
 PASS - test_sem_thread2thread in 0.001 seconds
===================================================================
START - test_sem_take_timeout
 PASS - test_sem_take_timeout in 0.001 seconds
===================================================================
START - test_sem_take_timeout_fails
 PASS - test_sem_take_timeout_fails in 0.548 seconds
===================================================================
START - test_sem_give_from_isr
 PASS - test_sem_give_from_isr in 0.001 seconds
===================================================================
START - test_sem_count_get
 PASS - test_sem_count_get in 0.001 seconds
===================================================================
START - test_sem_measure_timeouts
 PASS - test_sem_measure_timeouts in 1.009 seconds
===================================================================
START - test_sem_reset_waiting
 PASS - test_sem_reset_waiting in 0.020 seconds
===================================================================
START - test_sem_measure_timeout_from_thread
 PASS - test_sem_measure_timeout_from_thread in 0.001 seconds
===================================================================
START - test_k_sem_correct_count_limit
 PASS - test_k_sem_correct_count_limit in 0.001 seconds
===================================================================
START - test_k_sem_init
 PASS - test_k_sem_init in 0.001 seconds
===================================================================
START - test_k_sem_define
 PASS - test_k_sem_define in 0.001 seconds
===================================================================
START - test_sem_give_from_thread
 PASS - test_sem_give_from_thread in 0.001 seconds
===================================================================
START - test_sem_take_timeout_forever
 PASS - test_sem_take_timeout_forever in 0.107 seconds
===================================================================
START - test_sem_take_no_wait
 PASS - test_sem_take_no_wait in 0.001 seconds
===================================================================
START - test_sem_give_take_from_isr
 PASS - test_sem_give_take_from_isr in 0.001 seconds
===================================================================
START - test_sem_multiple_threads_wait
 PASS - test_sem_multiple_threads_wait in 2.038 seconds
===================================================================
START - test_sem_take_multiple
 PASS - test_sem_take_multiple in 1.010 seconds
===================================================================
START - test_sem_give_take_from_isr
 PASS - test_sem_give_take_from_isr in 0.001 seconds
===================================================================
START - test_sem_take_timeout_isr
 PASS - test_sem_take_timeout_isr in 0.001 seconds
===================================================================
START - test_sem_thread2thread
 PASS - test_sem_thread2thread in 0.001 seconds
===================================================================
START - test_sem_take_timeout
 PASS - test_sem_take_timeout in 0.001 seconds
===================================================================
START - test_k_sem_init
 PASS - test_k_sem_init in 0.001 seconds
===================================================================
START - test_sem_reset_waiting
 PASS - test_sem_reset_waiting in 0.016 seconds
===================================================================
START - test_sem_give_from_isr
 PASS - test_sem_give_from_isr in 0.001 seconds
===================================================================
START - test_sem_measure_timeout_from_thread
 PASS - test_sem_measure_timeout_from_thread in 0.001 seconds
===================================================================
START - test_k_sem_define
 PASS - test_k_sem_define in 0.001 seconds
===================================================================
START - test_sem_thread2isr
 PASS - test_sem_thread2isr in 0.001 seconds
===================================================================
START - test_sem_count_get
 PASS - test_sem_count_get in 0.001 seconds
===================================================================
START - test_sem_take_timeout_forever
 PASS - test_sem_take_timeout_forever in 0.107 seconds
===================================================================
START - test_sem_measure_timeouts
 PASS - test_sem_measure_timeouts in 1.010 seconds
===================================================================
START - test_sem_reset
 PASS - test_sem_reset in 0.110 seconds
===================================================================
START - test_sem_take_no_wait_fails
 PASS - test_sem_take_no_wait_fails in 0.001 seconds
===================================================================
START - test_sem_take_no_wait
 PASS - test_sem_take_no_wait in 0.001 seconds
===================================================================
START - test_k_sem_correct_count_limit
 PASS - test_k_sem_correct_count_limit in 0.001 seconds
===================================================================
START - test_sem_give_from_thread
 PASS - test_sem_give_from_thread in 0.001 seconds
===================================================================
START - test_sem_take_multiple
 PASS - test_sem_take_multiple in 1.007 seconds
===================================================================
START - test_sem_multi_take_timeout_diff_sem
 PASS - test_sem_multi_take_timeout_diff_sem in 5.010 seconds
===================================================================
START - test_sem_take_timeout_fails
 PASS - test_sem_take_timeout_fails in 0.550 seconds
===================================================================
START - test_sem_multiple_threads_wait
 PASS - test_sem_multiple_threads_wait in 2.040 seconds
===================================================================
TESTSUITE semaphore succeeded
Running TESTSUITE semaphore
===================================================================
START - test_k_sem_init
 PASS - test_k_sem_init in 0.001 seconds
===================================================================
START - test_sem_take_timeout
 PASS - test_sem_take_timeout in 0.001 seconds
===================================================================
START - test_sem_take_timeout_fails
 PASS - test_sem_take_timeout_fails in 0.548 seconds
===================================================================
START - test_sem_give_take_from_isr
 PASS - test_sem_give_take_from_isr in 0.001 seconds
===================================================================
START - test_sem_measure_timeouts
 PASS - test_sem_measure_timeouts in 1.009 seconds
===================================================================
START - test_sem_take_multiple
 PASS - test_sem_take_multiple in 1.010 seconds
===================================================================
START - test_sem_multi_take_timeout_diff_sem
 PASS - test_sem_multi_take_timeout_diff_sem in 5.010 seconds
===================================================================
START - test_sem_take_no_wait
 PASS - test_sem_take_no_wait in 0.001 seconds
===================================================================
START - test_k_sem_correct_count_limit
 PASS - test_k_sem_correct_count_limit in 0.001 seconds
===================================================================
START - test_sem_thread2thread
 PASS - test_sem_thread2thread in 0.001 seconds
===================================================================
START - test_sem_give_from_thread
 PASS - test_sem_give_from_thread in 0.001 seconds
===================================================================
START - test_sem_take_timeout_isr
 PASS - test_sem_take_timeout_isr in 0.001 seconds
===================================================================
START - test_sem_take_timeout_forever
 PASS - test_sem_take_timeout_forever in 0.106 seconds
===================================================================
START - test_sem_multiple_threads_wait
 PASS - test_sem_multiple_threads_wait in 2.040 seconds
===================================================================
START - test_sem_reset_waiting
 PASS - test_sem_reset_waiting in 0.020 seconds
===================================================================
START - test_sem_take_no_wait_fails
 PASS - test_sem_take_no_wait_fails in 0.001 seconds
===================================================================
START - test_sem_reset
 PASS - test_sem_reset in 0.109 seconds
===================================================================
START - test_sem_give_from_isr
 PASS - test_sem_give_from_isr in 0.001 seconds
===================================================================
START - test_sem_measure_timeout_from_thread
 PASS - test_sem_measure_timeout_from_thread in 0.001 seconds
===================================================================
START - test_k_sem_define
 PASS - test_k_sem_define in 0.001 seconds
===================================================================
START - test_sem_thread2isr
 PASS - test_sem_thread2isr in 0.001 seconds
===================================================================
START - test_sem_count_get
 PASS - test_sem_count_get in 0.001 seconds
===================================================================
START - test_sem_multi_take_timeout_diff_sem
 PASS - test_sem_multi_take_timeout_diff_sem in 5.006 seconds
===================================================================
START - test_sem_give_from_thread
 PASS - test_sem_give_from_thread in 0.001 seconds
===================================================================
START - test_sem_measure_timeout_from_thread
 PASS - test_sem_measure_timeout_from_thread in 0.001 seconds
===================================================================
START - test_sem_take_timeout
 PASS - test_sem_take_timeout in 0.001 seconds
===================================================================
START - test_sem_multiple_threads_wait
 PASS - test_sem_multiple_threads_wait in 2.038 seconds
===================================================================
START - test_sem_measure_timeouts
 PASS - test_sem_measure_timeouts in 1.010 seconds
===================================================================
START - test_sem_thread2thread
 PASS - test_sem_thread2thread in 0.001 seconds
===================================================================
START - test_k_sem_init
 PASS - test_k_sem_init in 0.001 seconds
===================================================================
START - test_sem_give_from_isr
 PASS - test_sem_give_from_isr in 0.001 seconds
===================================================================
START - test_sem_take_timeout_isr
 PASS - test_sem_take_timeout_isr in 0.001 seconds
===================================================================
START - test_k_sem_correct_count_limit
 PASS - test_k_sem_correct_count_limit in 0.001 seconds
===================================================================
START - test_sem_count_get
 PASS - test_sem_count_get in 0.001 seconds
===================================================================
START - test_sem_thread2isr
 PASS - test_sem_thread2isr in 0.001 seconds
===================================================================
START - test_sem_take_multiple
 PASS - test_sem_take_multiple in 1.005 seconds
===================================================================
START - test_sem_take_no_wait_fails
 PASS - test_sem_take_no_wait_fails in 0.001 seconds
===================================================================
START - test_sem_reset
 PASS - test_sem_reset in 0.109 seconds
===================================================================
START - test_sem_take_no_wait
 PASS - test_sem_take_no_wait in 0.001 seconds
===================================================================
START - test_sem_reset_waiting
 PASS - test_sem_reset_waiting in 0.019 seconds
===================================================================
START - test_k_sem_define
 PASS - test_k_sem_define in 0.001 seconds
===================================================================
START - test_sem_take_timeout_forever
 PASS - test_sem_take_timeout_forever in 0.109 seconds
===================================================================
START - test_sem_give_take_from_isr
 PASS - test_sem_give_take_from_isr in 0.001 seconds
===================================================================
START - test_sem_take_timeout_fails
 PASS - test_sem_take_timeout_fails in 0.549 seconds
===================================================================
START - test_k_sem_define
 PASS - test_k_sem_define in 0.001 seconds
===================================================================
START - test_sem_multiple_threads_wait
 PASS - test_sem_multiple_threads_wait in 2.039 seconds
===================================================================
START - test_sem_thread2isr
 PASS - test_sem_thread2isr in 0.001 seconds
===================================================================
START - test_sem_take_timeout_forever
 PASS - test_sem_take_timeout_forever in 0.109 seconds
===================================================================
START - test_sem_give_take_from_isr
 PASS - test_sem_give_take_from_isr in 0.001 seconds
===================================================================
START - test_sem_take_multiple
 PASS - test_sem_take_multiple in 1.009 seconds
===================================================================
START - test_sem_take_timeout
 PASS - test_sem_take_timeout in 0.001 seconds
===================================================================
START - test_sem_take_timeout_fails
 PASS - test_sem_take_timeout_fails in 0.549 seconds
===================================================================
START - test_k_sem_init
 PASS - test_k_sem_init in 0.001 seconds
===================================================================
START - test_sem_reset_waiting
 PASS - test_sem_reset_waiting in 0.020 seconds
===================================================================
START - test_sem_thread2thread
 PASS - test_sem_thread2thread in 0.001 seconds
===================================================================
START - test_sem_reset
 PASS - test_sem_reset in 0.109 seconds
===================================================================
START - test_sem_take_timeout_isr
 PASS - test_sem_take_timeout_isr in 0.001 seconds
===================================================================
START - test_sem_measure_timeouts
 PASS - test_sem_measure_timeouts in 1.009 seconds
===================================================================
START - test_sem_take_no_wait
 PASS - test_sem_take_no_wait in 0.001 seconds
===================================================================
START - test_sem_measure_timeout_from_thread
 PASS - test_sem_measure_timeout_from_thread in 0.001 seconds
===================================================================
START - test_k_sem_correct_count_limit
 PASS - test_k_sem_correct_count_limit in 0.001 seconds
===================================================================
START - test_sem_give_from_thread
 PASS - test_sem_give_from_thread in 0.001 seconds
===================================================================
START - test_sem_take_no_wait_fails
 PASS - test_sem_take_no_wait_fails in 0.001 seconds
===================================================================
START - test_sem_count_get
 PASS - test_sem_count_get in 0.001 seconds
===================================================================
START - test_sem_give_from_isr
 PASS - test_sem_give_from_isr in 0.001 seconds
===================================================================
START - test_sem_multi_take_timeout_diff_sem
 PASS - test_sem_multi_take_timeout_diff_sem in 5.005 seconds
===================================================================
TESTSUITE semaphore succeeded

------ TESTSUITE SUMMARY START ------

SUITE PASS - 100.00% [semaphore]: pass = 22, fail = 0, skip = 0, total = 22 duration = 9.871 seconds
 - PASS - [semaphore.test_k_sem_correct_count_limit] duration = 0.001 seconds
 - PASS - [semaphore.test_k_sem_define] duration = 0.001 seconds
 - PASS - [semaphore.test_k_sem_init] duration = 0.001 seconds
 - PASS - [semaphore.test_sem_count_get] duration = 0.001 seconds
 - PASS - [semaphore.test_sem_give_from_isr] duration = 0.001 seconds
 - PASS - [semaphore.test_sem_give_from_thread] duration = 0.001 seconds
 - PASS - [semaphore.test_sem_give_take_from_isr] duration = 0.001 seconds
 - PASS - [semaphore.test_sem_measure_timeout_from_thread] duration = 0.001 seconds
 - PASS - [semaphore.test_sem_measure_timeouts] duration = 1.010 seconds
 - PASS - [semaphore.test_sem_multi_take_timeout_diff_sem] duration = 5.010 seconds
 - PASS - [semaphore.test_sem_multiple_threads_wait] duration = 2.040 seconds
 - PASS - [semaphore.test_sem_reset] duration = 0.109 seconds
 - PASS - [semaphore.test_sem_reset_waiting] duration = 0.020 seconds
 - PASS - [semaphore.test_sem_take_multiple] duration = 1.010 seconds
 - PASS - [semaphore.test_sem_take_no_wait] duration = 0.001 seconds
 - PASS - [semaphore.test_sem_take_no_wait_fails] duration = 0.001 seconds
 - PASS - [semaphore.test_sem_take_timeout] duration = 0.001 seconds
 - PASS - [semaphore.test_sem_take_timeout_fails] duration = 0.549 seconds
 - PASS - [semaphore.test_sem_take_timeout_forever] duration = 0.109 seconds
 - PASS - [semaphore.test_sem_take_timeout_isr] duration = 0.001 seconds
 - PASS - [semaphore.test_sem_thread2isr] duration = 0.001 seconds
 - PASS - [semaphore.test_sem_thread2thread] duration = 0.001 seconds

SUITE PASS - 100.00% [semaphore_1cpu]: pass = 2, fail = 0, skip = 0, total = 2 duration = 1.454 seconds
 - PASS - [semaphore_1cpu.test_sem_multiple_take_and_timeouts] duration = 1.260 seconds
 - PASS - [semaphore_1cpu.test_sem_queue_mutual_exclusion] duration = 0.194 seconds

SUITE SKIP -   0.00% [semaphore_null_case]: pass = 0, fail = 0, skip = 0, total = 0 duration = 0.000 seconds

------ TESTSUITE SUMMARY END ------

===================================================================
PROJECT EXECUTION FAILED
```

with the change
```
-- west build: running target run
[0/1] To exit from QEMU enter: 'CTRL+a, x'[QEMU] CPU: cortex-m3
qemu-system-arm: warning: nic stellaris_enet.0 has no peer
Timer with period zero, disabling
*** Booting Zephyr OS build zephyr-v3.4.0-1607-ga98d2c3aafce ***
Running TESTSUITE semaphore_1cpu
===================================================================
START - test_sem_multiple_take_and_timeouts
 PASS - test_sem_multiple_take_and_timeouts in 1.260 seconds
===================================================================
START - test_sem_queue_mutual_exclusion
 PASS - test_sem_queue_mutual_exclusion in 0.194 seconds
===================================================================
START - test_sem_multiple_take_and_timeouts
 PASS - test_sem_multiple_take_and_timeouts in 1.256 seconds
===================================================================
START - test_sem_queue_mutual_exclusion
 PASS - test_sem_queue_mutual_exclusion in 0.194 seconds
===================================================================
START - test_sem_queue_mutual_exclusion
 PASS - test_sem_queue_mutual_exclusion in 0.190 seconds
===================================================================
START - test_sem_multiple_take_and_timeouts
 PASS - test_sem_multiple_take_and_timeouts in 1.256 seconds
===================================================================
TESTSUITE semaphore_1cpu succeeded
Running TESTSUITE semaphore_1cpu
===================================================================
START - test_sem_queue_mutual_exclusion
 PASS - test_sem_queue_mutual_exclusion in 0.193 seconds
===================================================================
START - test_sem_multiple_take_and_timeouts
 PASS - test_sem_multiple_take_and_timeouts in 1.256 seconds
===================================================================
START - test_sem_queue_mutual_exclusion
 PASS - test_sem_queue_mutual_exclusion in 0.193 seconds
===================================================================
START - test_sem_multiple_take_and_timeouts
 PASS - test_sem_multiple_take_and_timeouts in 1.256 seconds
===================================================================
START - test_sem_multiple_take_and_timeouts
 PASS - test_sem_multiple_take_and_timeouts in 1.260 seconds
===================================================================
START - test_sem_queue_mutual_exclusion
 PASS - test_sem_queue_mutual_exclusion in 0.194 seconds
===================================================================
TESTSUITE semaphore_1cpu succeeded
Running TESTSUITE semaphore_1cpu
===================================================================
START - test_sem_multiple_take_and_timeouts
 PASS - test_sem_multiple_take_and_timeouts in 1.255 seconds
===================================================================
START - test_sem_queue_mutual_exclusion
 PASS - test_sem_queue_mutual_exclusion in 0.194 seconds
===================================================================
START - test_sem_queue_mutual_exclusion
 PASS - test_sem_queue_mutual_exclusion in 0.190 seconds
===================================================================
START - test_sem_multiple_take_and_timeouts
 PASS - test_sem_multiple_take_and_timeouts in 1.256 seconds
===================================================================
START - test_sem_queue_mutual_exclusion
 PASS - test_sem_queue_mutual_exclusion in 0.194 seconds
===================================================================
START - test_sem_multiple_take_and_timeouts
 PASS - test_sem_multiple_take_and_timeouts in 1.256 seconds
===================================================================
TESTSUITE semaphore_1cpu succeeded
Running TESTSUITE semaphore_null_case
===================================================================
TESTSUITE semaphore_null_case succeeded
Running TESTSUITE semaphore_null_case
===================================================================
TESTSUITE semaphore_null_case succeeded
Running TESTSUITE semaphore_null_case
===================================================================
TESTSUITE semaphore_null_case succeeded
Running TESTSUITE semaphore
===================================================================
START - test_sem_take_timeout_isr
 PASS - test_sem_take_timeout_isr in 0.001 seconds
===================================================================
START - test_sem_take_timeout_forever
 PASS - test_sem_take_timeout_forever in 0.104 seconds
===================================================================
START - test_sem_take_timeout_fails
 PASS - test_sem_take_timeout_fails in 0.550 seconds
===================================================================
START - test_k_sem_define
 PASS - test_k_sem_define in 0.001 seconds
===================================================================
START - test_sem_thread2thread
 PASS - test_sem_thread2thread in 0.001 seconds
===================================================================
START - test_k_sem_init
 PASS - test_k_sem_init in 0.001 seconds
===================================================================
START - test_sem_multiple_threads_wait
 PASS - test_sem_multiple_threads_wait in 2.038 seconds
===================================================================
START - test_sem_thread2isr
 PASS - test_sem_thread2isr in 0.001 seconds
===================================================================
START - test_sem_give_take_from_isr
 PASS - test_sem_give_take_from_isr in 0.001 seconds
===================================================================
START - test_sem_reset_waiting
 PASS - test_sem_reset_waiting in 0.018 seconds
===================================================================
START - test_sem_multi_take_timeout_diff_sem
 PASS - test_sem_multi_take_timeout_diff_sem in 5.010 seconds
===================================================================
START - test_sem_give_from_isr
 PASS - test_sem_give_from_isr in 0.001 seconds
===================================================================
START - test_sem_take_multiple
 PASS - test_sem_take_multiple in 1.009 seconds
===================================================================
START - test_sem_give_from_thread
 PASS - test_sem_give_from_thread in 0.001 seconds
===================================================================
START - test_sem_measure_timeout_from_thread

    Assertion failed at WEST_TOPDIR/zephyr/tests/kernel/semaphore/semaphore/src/main.c:947: semaphore_test_sem_measure_timeout_from_thread: (1 is non-zero)
Forced error
 FAIL - test_sem_measure_timeout_from_thread in 0.001 seconds
===================================================================
START - test_sem_take_timeout
 PASS - test_sem_take_timeout in 0.001 seconds
===================================================================
START - test_sem_take_no_wait_fails
 PASS - test_sem_take_no_wait_fails in 0.001 seconds
===================================================================
START - test_sem_measure_timeouts
 PASS - test_sem_measure_timeouts in 1.007 seconds
===================================================================
START - test_sem_count_get
 PASS - test_sem_count_get in 0.001 seconds
===================================================================
START - test_sem_take_no_wait
 PASS - test_sem_take_no_wait in 0.001 seconds
===================================================================
START - test_k_sem_correct_count_limit
 PASS - test_k_sem_correct_count_limit in 0.001 seconds
===================================================================
START - test_sem_reset
 PASS - test_sem_reset in 0.108 seconds
===================================================================
START - test_sem_give_from_isr
 PASS - test_sem_give_from_isr in 0.001 seconds
===================================================================
START - test_sem_measure_timeouts
 PASS - test_sem_measure_timeouts in 1.009 seconds
===================================================================
START - test_sem_reset
 PASS - test_sem_reset in 0.110 seconds
===================================================================
START - test_sem_multi_take_timeout_diff_sem
 PASS - test_sem_multi_take_timeout_diff_sem in 5.010 seconds
===================================================================
START - test_sem_take_timeout_forever
 PASS - test_sem_take_timeout_forever in 0.110 seconds
===================================================================
START - test_sem_reset_waiting
 PASS - test_sem_reset_waiting in 0.020 seconds
===================================================================
START - test_sem_take_no_wait
 PASS - test_sem_take_no_wait in 0.001 seconds
===================================================================
START - test_sem_give_take_from_isr
 PASS - test_sem_give_take_from_isr in 0.001 seconds
===================================================================
START - test_sem_take_timeout
 PASS - test_sem_take_timeout in 0.001 seconds
===================================================================
START - test_sem_thread2isr
 PASS - test_sem_thread2isr in 0.001 seconds
===================================================================
START - test_k_sem_init
 PASS - test_k_sem_init in 0.001 seconds
===================================================================
START - test_sem_take_timeout_fails
 PASS - test_sem_take_timeout_fails in 0.546 seconds
===================================================================
START - test_sem_give_from_thread
 PASS - test_sem_give_from_thread in 0.001 seconds
===================================================================
START - test_sem_take_no_wait_fails
 PASS - test_sem_take_no_wait_fails in 0.001 seconds
===================================================================
START - test_sem_measure_timeout_from_thread
 PASS - test_sem_measure_timeout_from_thread in 0.001 seconds
===================================================================
START - test_sem_take_timeout_isr
 PASS - test_sem_take_timeout_isr in 0.001 seconds
===================================================================
START - test_k_sem_define
 PASS - test_k_sem_define in 0.001 seconds
===================================================================
START - test_k_sem_correct_count_limit
 PASS - test_k_sem_correct_count_limit in 0.001 seconds
===================================================================
START - test_sem_count_get
 PASS - test_sem_count_get in 0.001 seconds
===================================================================
START - test_sem_multiple_threads_wait
 PASS - test_sem_multiple_threads_wait in 2.035 seconds
===================================================================
START - test_sem_take_multiple
 PASS - test_sem_take_multiple in 1.010 seconds
===================================================================
START - test_sem_thread2thread
 PASS - test_sem_thread2thread in 0.001 seconds
===================================================================
START - test_sem_reset
 PASS - test_sem_reset in 0.109 seconds
===================================================================
START - test_sem_thread2thread
 PASS - test_sem_thread2thread in 0.001 seconds
===================================================================
START - test_sem_take_multiple
 PASS - test_sem_take_multiple in 1.009 seconds
===================================================================
START - test_sem_measure_timeout_from_thread
 PASS - test_sem_measure_timeout_from_thread in 0.001 seconds
===================================================================
START - test_sem_take_no_wait_fails
 PASS - test_sem_take_no_wait_fails in 0.001 seconds
===================================================================
START - test_k_sem_correct_count_limit
 PASS - test_k_sem_correct_count_limit in 0.001 seconds
===================================================================
START - test_sem_take_timeout_isr
 PASS - test_sem_take_timeout_isr in 0.001 seconds
===================================================================
START - test_sem_multi_take_timeout_diff_sem
 PASS - test_sem_multi_take_timeout_diff_sem in 5.007 seconds
===================================================================
START - test_sem_reset_waiting
 PASS - test_sem_reset_waiting in 0.020 seconds
===================================================================
START - test_sem_give_from_isr
 PASS - test_sem_give_from_isr in 0.001 seconds
===================================================================
START - test_sem_take_timeout_fails
 PASS - test_sem_take_timeout_fails in 0.549 seconds
===================================================================
START - test_sem_count_get
 PASS - test_sem_count_get in 0.001 seconds
===================================================================
START - test_sem_take_no_wait
 PASS - test_sem_take_no_wait in 0.001 seconds
===================================================================
START - test_sem_give_take_from_isr
 PASS - test_sem_give_take_from_isr in 0.001 seconds
===================================================================
START - test_sem_take_timeout_forever
 PASS - test_sem_take_timeout_forever in 0.108 seconds
===================================================================
START - test_k_sem_define
 PASS - test_k_sem_define in 0.001 seconds
===================================================================
START - test_sem_measure_timeouts
 PASS - test_sem_measure_timeouts in 1.009 seconds
===================================================================
START - test_sem_multiple_threads_wait
 PASS - test_sem_multiple_threads_wait in 2.040 seconds
===================================================================
START - test_sem_take_timeout
 PASS - test_sem_take_timeout in 0.001 seconds
===================================================================
START - test_k_sem_init
 PASS - test_k_sem_init in 0.001 seconds
===================================================================
START - test_sem_thread2isr
 PASS - test_sem_thread2isr in 0.001 seconds
===================================================================
START - test_sem_give_from_thread
 PASS - test_sem_give_from_thread in 0.001 seconds
===================================================================
TESTSUITE semaphore succeeded

------ TESTSUITE SUMMARY START ------

SUITE SKIP -   0.00% [semaphore]: pass = 0, fail = 0, skip = 22, total = 22 duration = 0.000 seconds
 - SKIP - [semaphore.test_k_sem_correct_count_limit] duration = 0.000 seconds
 - SKIP - [semaphore.test_k_sem_define] duration = 0.000 seconds
 - SKIP - [semaphore.test_k_sem_init] duration = 0.000 seconds
 - SKIP - [semaphore.test_sem_count_get] duration = 0.000 seconds
 - SKIP - [semaphore.test_sem_give_from_isr] duration = 0.000 seconds
 - SKIP - [semaphore.test_sem_give_from_thread] duration = 0.000 seconds
 - SKIP - [semaphore.test_sem_give_take_from_isr] duration = 0.000 seconds
 - SKIP - [semaphore.test_sem_measure_timeout_from_thread] duration = 0.000 seconds
 - SKIP - [semaphore.test_sem_measure_timeouts] duration = 0.000 seconds
 - SKIP - [semaphore.test_sem_multi_take_timeout_diff_sem] duration = 0.000 seconds
 - SKIP - [semaphore.test_sem_multiple_threads_wait] duration = 0.000 seconds
 - SKIP - [semaphore.test_sem_reset] duration = 0.000 seconds
 - SKIP - [semaphore.test_sem_reset_waiting] duration = 0.000 seconds
 - SKIP - [semaphore.test_sem_take_multiple] duration = 0.000 seconds
 - SKIP - [semaphore.test_sem_take_no_wait] duration = 0.000 seconds
 - SKIP - [semaphore.test_sem_take_no_wait_fails] duration = 0.000 seconds
 - SKIP - [semaphore.test_sem_take_timeout] duration = 0.000 seconds
 - SKIP - [semaphore.test_sem_take_timeout_fails] duration = 0.000 seconds
 - SKIP - [semaphore.test_sem_take_timeout_forever] duration = 0.000 seconds
 - SKIP - [semaphore.test_sem_take_timeout_isr] duration = 0.000 seconds
 - SKIP - [semaphore.test_sem_thread2isr] duration = 0.000 seconds
 - SKIP - [semaphore.test_sem_thread2thread] duration = 0.000 seconds

SUITE PASS - 100.00% [semaphore_1cpu]: pass = 2, fail = 0, skip = 0, total = 2 duration = 1.450 seconds
 - PASS - [semaphore_1cpu.test_sem_multiple_take_and_timeouts] duration = 1.256 seconds
 - PASS - [semaphore_1cpu.test_sem_queue_mutual_exclusion] duration = 0.194 seconds

SUITE SKIP -   0.00% [semaphore_null_case]: pass = 0, fail = 0, skip = 0, total = 0 duration = 0.000 seconds

------ TESTSUITE SUMMARY END ------

===================================================================
PROJECT EXECUTION FAILED
```